katharsis-client/src/main/java/io/katharsis/client/internal/AbstractStub.java

```diff
@@ -15,120 +13,102 @@
 import io.katharsis.client.http.HttpAdapterRequest;
 import io.katharsis.client.http.HttpAdapterResponse;
 import io.katharsis.dispatcher.controller.HttpMethod;
-import io.katharsis.errorhandling.ErrorData;
 import io.katharsis.errorhandling.ErrorResponse;
 import io.katharsis.errorhandling.mapper.ExceptionMapper;
 import io.katharsis.errorhandling.mapper.ExceptionMapperRegistry;
-import io.katharsis.resource.list.DefaultResourceList;
-import io.katharsis.response.BaseResponseContext;
-import io.katharsis.response.JsonApiResponse;
-import io.katharsis.response.LinksInformation;
-import io.katharsis.response.MetaInformation;
+import io.katharsis.resource.Document;
 import io.katharsis.utils.JsonApiUrlBuilder;
 import io.katharsis.utils.java.Optional;
 
 public class AbstractStub {
-	
+
 	private static final Logger LOGGER = LoggerFactory.getLogger(AbstractStub.class);
 
 	public static final String CONTENT_TYPE = "application/vnd.api+json";
 
-	protected KatharsisClient katharsis;
+	protected KatharsisClient client;
 
 	protected JsonApiUrlBuilder urlBuilder;
 
 	public AbstractStub(KatharsisClient client, JsonApiUrlBuilder urlBuilder) {
```

Comment:
Use `@RequiredArgsConstructor` on the class and declare the fields final. Then, get rid of this constructor